### DISCUSSION TITLE
Bounding boxs around atoms and bonds in image

### DISCUSSION THREAD

**Discussion by torphix** [rdkit/rdkit#3991] [Q&A]
Hello! 
I was wondering if its possible to draw bounding boxes around the various atoms and bonds within a molecules and then save the coordinates of the bounding box to a numpy array.
Thank you kindly

**Comment by DavidACosgrove** [comment]
Hi,
That functionality isn’t exposed in the public interface. If you’re feeling
brave enough to tackle the c++, the code to draw boxes round the characters
is there but commented out. I find it very convenient when debugging the
drawing code.

Best,
Dave


On Tue, 30 Mar 2021 at 13:17, Jaixy ***@***.***> wrote:

> Hello!
> I was wondering if its possible to draw bounding boxes around the various
> atoms and bonds within a molecules and then save the coordinates of the
> bounding box to a numpy array.
> Thank you kindly
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/3991>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FR4EHDH4LV2HPUO74DTGG6PDANCNFSM42B5B3HQ>
> .
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Comment by torphix** [comment]
Thank you kindly Dave!
I'll have a look for sure.
Is it perhaps possible to obtain the pixel coordinates of various atoms and
then infer bounding box's from the coordinates?
Thanks again

**Comment by DavidACosgrove** [comment]
The C++ code does have a way of getting the pixel coords from the atom
coords, though I can’t recall the syntax offhand. I’ve never had call to
find out if it’s exposed in Python.
Dave


On Tue, 30 Mar 2021 at 17:40, Jaixy ***@***.***> wrote:

> Thank you kindly Dave!
> I'll have a look for sure.
> Is it perhaps possible to obtain the pixel coordinates of various atoms and
> then infer bounding box's from the coordinates?
> Thanks again
>
> On Tue, 30 Mar 2021, 16:38 David Cosgrove, ***@***.***> wrote:
>
> > Hi,
> > That functionality isn’t exposed in the public interface. If you’re
> feeling
> > brave enough to tackle the c++, the code to draw boxes round the
> characters
> > is there but commented out. I find it very convenient when debugging the
> > drawing code.
> >
> > Best,
> > Dave
> >
> >
> > On Tue, 30 Mar 2021 at 13:17, Jaixy ***@***.***> wrote:
> >
> > > Hello!
> > > I was wondering if its possible to draw bounding boxes around the
> various
> > > atoms and bonds within a molecules and then save the coordinates of the
> > > bounding box to a numpy array.
> > > Thank you kindly
> > >
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/rdkit/rdkit/discussions/3991>, or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ACGF2FR4EHDH4LV2HPUO74DTGG6PDANCNFSM42B5B3HQ
> > >
> > > .
> > >
> > --
> > David Cosgrove
> > Freelance computational chemistry and chemoinformatics developer
> > http://cozchemix.co.uk
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/rdkit/rdkit/discussions/3991#discussioncomment-548787>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AP52P62TZ2VMJOTXXHLED2LTGHV5PANCNFSM42B5B3HQ
> >
> > .
> >
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/3991#discussioncomment-549257>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FQ6LZIG7B4MLDL5TG3TGH5GHANCNFSM42B5B3HQ>
> .
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
